### PR TITLE
[OpenAI-Assistant] Added vector store feature

### PR DESCRIPTION
**Pull request by mssfang**
- Added support for vector store, vector store with file and a batch of files operations.

### CODE REVIEW

File: sdk/openai/azure-ai-openai-assistants/README.md [hunk 1]
```diff
@@ -167,8 +167,8 @@ an appropriate tool like Code Interpreter or Retrieval is enabled.
             new AssistantCreationOptions(deploymentOrModelId)
                 .setName("Java SDK Retrieval Sample")
                 .setInstructions("You are a helpful assistant that can help fetch data from files you know about.")
-                .setTools(Arrays.asList(new RetrievalToolDefinition()))
-                // TODO - setup with VectorStore
+                .setTools(Arrays.asList(new FileSearchToolDefinition()))
+                // TODO (jose): setup with VectorStore
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/README.md:171]: This is fixed in the base feature branch 👍

File: sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/models/AssistantsApiResponseFormatOption.java [hunk 2]
```diff
@@ -71,6 +74,7 @@ public AssistantsApiResponseFormat getFormat() {
      *
      * @param responseFormatBinaryData input JSON string
      * @return a new instance of AssistantsApiResponseFormatOption
+     * @throws IllegalArgumentException if the provided JSON string does not match the expected format
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/models/AssistantsApiResponseFormatOption.java:77]: I think you might get some merge conflicts here, but they should be easy to address I hope. Sorry for giving you such a half-baked base feature branch 😓

File: sdk/openai/azure-ai-openai-assistants/src/main/resources/META-INF/azure-ai-openai-assistants_apiview_properties.json [hunk 3]
```diff
@@ -165,6 +165,7 @@
     "com.azure.ai.openai.assistants.AssistantsClient.uploadFileWithResponse": "Azure.AI.OpenAI.Assistants.uploadFile",
     "com.azure.ai.openai.assistants.AssistantsClientBuilder": "Azure.AI.OpenAI.Assistants.AssistantsClient",
     "com.azure.ai.openai.assistants.implementation.models.CreateVectorStoreFileBatchRequest": "",
+    "com.azure.ai.openai.assistants.implementation.models.CreateVectorStoreFileRequest": "",
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/src/main/resources/META-INF/azure-ai-openai-assistants_apiview_properties.json:168]: Is this empty model addressed by your fix in the TSP repo?

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+            StepVerifier.create(client.createVectorStore(vectorStoreDetails))
+                    .assertNext(vectorStore -> {
+                        assertNotNull(vectorStore);
+                        vectorStoreId.set(vectorStore.getId());
+                        assertNotNull(vectorStore.getId());
+                    })
+                    .verifyComplete();
+
+            // clean up the created vector store
+//            deleteVectorStores(client, vectorStoreId.get());
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java:45]: should this be uncommented?

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+        createVectorStoreWithFileAsyncRunner((storeId, fileId) -> {
+            StepVerifier.create(client.createVectorStoreFile(storeId, fileId))
+                    .assertNext(vectorStoreFile -> {
+                        assertNotNull(vectorStoreFile);
+                        assertNotNull(vectorStoreFile.getId());
+                    })
+                    .verifyComplete();
+            // clean up the created vector store
+            deleteVectorStores(client, storeId);
+            client.deleteFile(fileId).block();
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java:137]: I was thinking that some of these calls could be pushed to the test runners. Just an idea for one of the many things we can refactor "when we have time"

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/VectorStoreAsyncTests.java [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+                        assertNotNull(vectorStoreFileBatchResponse);
+                        assertEquals(storeId, vectorStoreFileBatchResponse.getVectorStoreId());
+                        assertEquals(batchId, vectorStoreFileBatchResponse.getId());
+                        assertEquals(totalFileCounts, vectorStoreFileBatchResponse.getFileCounts().getTotal());
+                    })
+                    .verifyComplete();
+
+            // clean up the created vector store
+            deleteVectorStores(client, storeId);
+            // TODO: delete the files
```
Comment (jpalvarezl) [inline sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/VectorStoreAsyncTests.java:246]: left over TODO

**Review by jpalvarezl — APPROVED** [review]
I think it looks good. Apologies again for the likely conflicts you might get while rebasing or merging the base branch. 

Were you able to generate recordings for these tests? I don't see an update to the `assets.json` file, so I suppose you either encountered an issue or you wanted to avoid possible conflicts (which I totally understand)

**Comment by mssfang** [comment]
> CreateVectorStoreFileRequest

No worries. I am able to record the new tests' json file and works fine in playback mode now. 
Will address the unresolve feebacks soon

File: sdk/openai/azure-ai-openai-assistants/src/main/resources/META-INF/azure-ai-openai-assistants_apiview_properties.json (hunk 3 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai-assistants/src/main/resources/META-INF/azure-ai-openai-assistants_apiview_properties.json:168]: It is generated by the TSP fix PR and you can see it looks similar to CreateVectorStoreFileBatchRequest

File: sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java (hunk 4 above)
Comment (mssfang) [reply sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureVectorStoreAsyncTests.java:45]: Yeah. Will need to refactor a bit to clean up the resource after test done

**Pull request merged**